File: java-poc/src/main/java/io/craigmiller160/querydslpoc/java/data/mapper/PositionMapper.java
package io.craigmiller160.querydslpoc.java.data.mapper;

import io.craigmiller160.querydslpoc.java.data.dto.PositionDTO;
import io.craigmiller160.querydslpoc.java.data.entity.Position;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface PositionMapper {
  PositionDTO positionToDTO(final Position position);
}

File: kotlin-poc/src/main/kotlin/io/craigmiller160/querydslpoc/kotlin/data/mapper/PositionMapper.kt
package io.craigmiller160.querydslpoc.kotlin.data.mapper

import io.craigmiller160.querydslpoc.kotlin.data.dto.PositionDTO
import io.craigmiller160.querydslpoc.kotlin.data.entity.Position
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface PositionMapper {
  fun positionToDTO(position: Position): PositionDTO
}
